include/seastar/http/client.hh

```diff
@@ -295,7 +295,16 @@ public:
      * \param as -- abort source that aborts the request
      * \param expected -- the optional expected reply status code, default is std::nullopt
      */
-    future<> make_request(request req, reply_handler handle, abort_source& as, std::optional<reply::status_type> expected = std::nullopt);
+    future<> make_request(request&& req, reply_handler&& handle, abort_source& as, std::optional<reply::status_type>&& expected = std::nullopt);
+
+    /**
+     * \brief Send the request and handle the response, same as \ref make_request()
+     *
+     *  @attention Note that the method does not take the ownership of the
+     * `request and the `handle`, it caller's responsibility the make sure they
+     * are referencing valid instances
+     */
+    future<> make_request(request& req, reply_handler& handle, abort_source& as, std::optional<reply::status_type> expected = std::nullopt);
```

Comment:
How about non-owning make_request that doesn't take abort-source argument?

Reply:
Added